aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -164,40 +164,32 @@ public S3FileIO(
       S3FileIOProperties s3FileIOProperties) {
     this.s3 = s3;
     this.s3Async = s3Async;
-    this.s3FileIOProperties = s3FileIOProperties;
```

Comment:
I think we need to go through deprecations for this

Reply:
yes I agree, this is also the only reason why we're exposing the underlying properties from `S3FileIOProperties` / `GCPProperties`. I'll open a separate PR to deprecate this